python/TestHarness/util.py

```diff
@@ -616,6 +616,112 @@ def checkLibtorchVersion(checks, test):
 
     return (checkVersion(checks, version_string, 'libtorch_version'), version_string)
 
+def getCapabilities(exe):
+    """
+    Get capabilities JSON and compare it to the required capabilities
+    """
+    output = runCommand("%s --show-capabilities" % exe)
+    try:
+        output = output.split('**START JSON DATA**\n')[1]
+        output = output.split('**END JSON DATA**\n')[0]
+        results = json.loads(output)
+    except IndexError:
+        return {}
+    except json.decoder.JSONDecodeError:
+        print(f'{exe} --show-capabilities, produced invalid JSON output')
+        sys.exit(1)
```

Comment:
```suggestion
    except json.decoder.JSONDecodeError as e:
        raise Exception(f'{exe} --show-capabilities, produced invalid JSON output') from e
```

I've started doing this, much better